homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -0,0 +1,165 @@
+"""Classes for voice assistant pipelines."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from enum import Enum
+import time
+from typing import Any
+
+from aiohttp import StreamReader
+
+from homeassistant.components import conversation, stt
+from homeassistant.core import Context, HomeAssistant
+
+DEFAULT_TIMEOUT = 30  # seconds
+
+
+@dataclass
+class PipelineRequest:
+    """Request to start a pipeline run."""
+
+    stt_audio: StreamReader | None
+    stt_metadata: stt.SpeechMetadata | None
+    stt_text: str | None = None
+    conversation_id: str | None = None
+
+
+class PipelineEventType(str, Enum):
+    """Event types emitted during a pipeline run."""
+
+    RUN_START = "run-start"
+    RUN_FINISH = "run-finish"
+    STT_START = "stt-start"
+    STT_FINISH = "stt-finish"
+    INTENT_START = "intent-start"
+    INTENT_FINISH = "intent-finish"
+    TTS_START = "tts-start"
+    TTS_FINISH = "tts-finish"
+    ERROR = "error"
+
+
+@dataclass
+class PipelineEvent:
+    """Events emitted during a pipeline run."""
+
+    type: PipelineEventType
+    data: dict[str, Any] | None = None
+    timestamp: int = field(default_factory=time.monotonic_ns)
```

Comment:
Let's change this to a `utcnow().isoformat()` with `from homeassistant.util.dt import utcnow`